projects/mysadna/src/app/services/firebase.service.ts

```diff
@@ -31,72 +26,26 @@ export class FirebaseService {
   getUserList(): Observable<User[]> {
     return this.userList.snapshotChanges().pipe(
       map(action => action.map(a => {
-        const firebaseElement = a.payload.doc.data() as FirebaseElement;
-
-        if (firebaseElement === undefined) {
-          // Element not found
-          return;
-        }
-
-        this.tempUser = this.convertFirebaseElementToUser(firebaseElement);
-
-        if (this.users.find(user => (
-          user.getUserId() === this.tempUser.getUserId()))) {
-          const userIndex = this.users.findIndex(user => (
-            user.getUserId() === this.tempUser.getUserId())
-          );
-          this.users[userIndex] = this.tempUser;
-        } else {
-          this.users.push(this.tempUser);
-        }
-        return this.tempUser;
+        const firebaseElement: FirebaseElement = a.payload.doc.data();
+        return User.deserializeBinary(this.getBinary(firebaseElement));
       }))
     );
   }
 
   getProjectList(): Observable<Project[]> {
     return this.projectList.snapshotChanges().pipe(
       map(action => action.map(a => {
-        const firebaseElement = a.payload.doc.data() as FirebaseElement;
-
-        if (firebaseElement === undefined) {
-          // Element not found
-          return;
-        }
-
-        this.tempProject = this.convertFirebaseElementToProject(firebaseElement);
-
-        if (this.projects.find(project => (
-          project.getProjectId() === this.tempProject.getProjectId()))) {
-          const projectIndex = this.projects.findIndex(project => (
-            project.getProjectId() === this.tempProject.getProjectId())
-            );
-          this.projects[projectIndex] = this.tempProject;
-        } else {
-          this.projects.push(this.tempProject);
-        }
-        return this.tempProject;
+        const firebaseElement: FirebaseElement = a.payload.doc.data();
+        return Project.deserializeBinary(this.getBinary(firebaseElement));
       }))
     );
   }
 
-  private convertFirebaseElementToUser(firebaseElement: FirebaseElement): User {
-    // Convert firebaseElement to binary
-    const binary: Uint8Array = this.encodingService
-      .decodeBase64StringToUint8Array(firebaseElement.proto);
-    // Convert binary to user
-    const user: User = User.deserializeBinary(binary);
-
-    return user;
-  }
-
-  private convertFirebaseElementToProject(firebaseElement: FirebaseElement): Project {
-    // Convert firebaseElement to binary
-    const binary: Uint8Array = this.encodingService
-      .decodeBase64StringToUint8Array(firebaseElement.proto);
-    // Convert binary to project
-    const project: Project = Project.deserializeBinary(binary);
-
-    return project;
+  // Converts firebaseElement to binary
+  private getBinary(firebaseElement: FirebaseElement): Uint8Array {
+    const binary: Uint8Array = this.encodingService.decodeBase64StringToUint8Array(
+      firebaseElement.proto
+    );
+    return binary;
```

Comment:
Here's what I meant by "to make it more DRY"

Reply:
I didn't even think of that way.